tez-dag/src/main/java/org/apache/tez/dag/app/dag/impl/TaskImpl.java

```diff
@@ -465,15 +465,19 @@ public TaskReport getReport() {
 
   @Override
   public TezCounters getCounters() {
-    TezCounters counters = new TezCounters();
-    counters.incrAllCounters(this.counters);
+    TezCounters tezCounters = null;
+    if (getVertex().isSpeculationEnabled()) {
+      tezCounters = new TezCounters();
+      tezCounters.incrAllCounters(this.counters);
+    }
     readLock.lock();
     try {
       TaskAttempt bestAttempt = selectBestAttempt();
-      if (bestAttempt != null) {
-        counters.incrAllCounters(bestAttempt.getCounters());
+      if (bestAttempt != null && tezCounters != null) {
```

Comment:
tezCounters != null only if speculation is enabled due to the recent patch, would it make sense to use that condition here for better readability? I guess it would clarify that we only use "this.counter" and this whole stuff in case of speculation
also, we're returning a non-null counter anyway, would this be clearer as:
```
TezCounters taskCounters = (bestAttempt != null) ? bestAttempt.getCounters() : TaskAttemptImpl.EMPTY_COUNTERS;

if (speculationEnabled){
  tezCounters.incrAllCounters(taskCounters); 
  return tezCounters;
}
return taskCounters;
```

Reply:
done